tracer/build/_build/CriticalPathAnalysis/CriticalPathAnalyzer.cs

```diff
@@ -82,11 +118,24 @@ static List<PipelineStage> GetStages(AbsolutePath rootDirectory, PipelineDefinit
         // the successors (stages that depend on us)
         foreach (var currentStage in stages)
         {
-            var stageDefinition = pipeline.Stages.First(x => x.Stage == currentStage.Id);
+            var stageDefinition = pipeline.Stages.FirstOrDefault(x => x.Stage == currentStage.Id);
+            if (stageDefinition is null)
+            {
+                throw new Exception("Could not find stage called " + currentStage.Id);
```

Comment:
Same as [below](https://github.com/DataDog/dd-trace-dotnet/pull/6844/files#r2040709109), I think it would make sense to use string interpolation

```suggestion
                throw new Exception($"Could not find stage called {currentStage.Id}. " );
```

Again, I might be missing something here 🙂

Reply:
Meh, when it's a single suffix like this I sometimes don't bother, but mostly for laziness rather than any principled reason 😅